internal/cli/ondemand_runner_apply.go

```diff
@@ -0,0 +1,264 @@
+					clierrors.Humanize(diag),
+					terminal.WithErrorStyle(),
+				)
+
+				return 1
+			}
+
+		case ".json":
+			od.ConfigFormat = pb.Project_JSON
+			_, diag := hcljson.Parse(data, "<waypoint-hcl>")
+			if diag.HasErrors() {
+				c.ui.Output(
+					"Syntax errors in file specified with -plugin-config: %s",
+					clierrors.Humanize(diag),
+					terminal.WithErrorStyle(),
+				)
+
+				return 1
+			}
+
+		default:
+			c.ui.Output(
+				"File specified via -plugin-config must end in '.hcl' or '.json'",
+				terminal.WithErrorStyle(),
+			)
+
+			return 1
+		}
+	}
+
+	od.OciUrl = c.flagOCIUrl
+	od.EnvironmentVariables = map[string]string{}
+	od.Default = c.flagDefault
+
+	for _, kv := range c.flagEnvVars {
+		idx := strings.IndexByte(kv, '=')
+		if idx != -1 {
+			od.EnvironmentVariables[kv[:idx]] = kv[idx+1:]
+		}
+	}
+
+	od.PluginType = c.flagPluginType
+
+	// Upsert
+	_, err := c.project.Client().UpsertOndemandRunner(ctx, &pb.UpsertOndemandRunnerRequest{
+		OndemandRunner: od,
+	})
+	if err != nil {
+		c.ui.Output(
+			"Error upserting ondemand runner config: %s", clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	if updated {
+		s.Update("Ondemand Runner configuration updated")
+	} else {
+		s.Update("Ondemand Runner configuration created")
+	}
+	s.Done()
+
+	return 0
+}
+
+func (c *OndemandRunnerApplyCommand) Flags() *flag.Sets {
+	return c.flagSet(0, func(sets *flag.Sets) {
+		f := sets.NewSet("Command Options")
+
+		f.StringVar(&flag.StringVar{
+			Name:    "id",
+			Target:  &c.flagId,
+			Default: "",
+			Usage:   "The id an existing ondemand runner to update.",
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:    "oci-url",
+			Target:  &c.flagOCIUrl,
+			Default: "hashicorp/waypoint:stable",
+			Usage:   "The url for the OCI image to launch for the ondemand runner.",
+		})
+
+		f.StringSliceVar(&flag.StringSliceVar{
+			Name:   "env-vars",
+			Target: &c.flagEnvVars,
+			Usage: "Environment variable to expose to the ondemand runner. Typically used to " +
+				"introduce configuration for the plugins that the runner will execute.",
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:    "plugin-type",
+			Target:  &c.flagPluginType,
+			Default: "",
+			Usage:   "The type of the plugin to launch for the ondemand runner, such as aws-ecs, kubernetes, etc.",
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:    "project-config",
+			Target:  &c.flagPluginConfig,
+			Default: "",
+			Usage: "Path to a an hcl file that contains the configuration for the plugin." +
+				"This is only necessary when the plugin's defaults need to be adjusted for " +
+				"the environment the plugin will launch the ondemand runner.",
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "default",
+			Target:  &c.flagDefault,
+			Default: false,
+			Usage: "Indicates that this ondemand runner should be used by any project that doesn't " +
+				"otherwise specify it's own ondemand runner.",
```

Comment:
Izaak _just_ fixed these 😂 
```suggestion
				"otherwise specify its own ondemand runner.",
```